### PR TITLE
Added CI

### PR DESCRIPTION
**Pull request by marcelometal**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  # instead of container networking so you access them via localhost rather
+  # than the service name
+  vm-job:
+    runs-on: ubuntu-latest
+
+    services:
+      mongodb:
+        image: mongo
+        ports:
+        # will assign a random free host port
```
Comment (kov) [inline .github/workflows/ci.yml]: I don't get this comment?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+          components: rustfmt, clippy
+      - name: Setup
+        run: make setup
+      - name: Run test
+        run: make test
+        env:
+          # use mongodb for the host here because we have specified a container
+          # for the job.  If we were running the job on the VM this would be
+          # localhost
+          MONGODB_HOST: mongodb
```
Comment (kov) [inline .github/workflows/ci.yml]: Do we need support for these env variables in the codebase?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (marcelometal) [reply .github/workflows/ci.yml]: WIP man... ignore it, pls ;) it was copy & paste

File: .github/workflows/ci.yml (hunk 2 above)
Comment (marcelometal) [reply .github/workflows/ci.yml]: WIP... ignore it, pls ;) it was copy & paste

**Pull request merged**